### PR TITLE
Java NDarray Safety

### PR DESCRIPTION
**Pull request by johnc1231**

### CODE REVIEW

File: hail/src/main/scala/is/hail/annotations/UnsafeRow.scala [hunk 1]
```diff
@@ -244,13 +244,16 @@ object SafeRow {
...
+      case _: UnsafeNDArray => false
 
       case r: Row =>
         r.toSeq.forall(isSafe)
       case a: IndexedSeq[_] =>
         a.forall(isSafe)
       case i: Interval =>
         isSafe(i.start) && isSafe(i.end)
+      case nd: NDArray =>
+        isSafe(nd.getRowMajorElements().forall(isSafe))
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/annotations/UnsafeRow.scala:256]: seems wrong -- should check type of top level and then recur to elements?

**Review by tpoterba — DISMISSED** [review]

File: hail/src/main/scala/is/hail/annotations/UnsafeRow.scala (hunk 1 above)
Comment (johnc1231) [reply hail/src/main/scala/is/hail/annotations/UnsafeRow.scala:256]: Top level is checked above right?

Comment (tpoterba) [reply hail/src/main/scala/is/hail/annotations/UnsafeRow.scala:256]: oh! I see.

Should be         

`nd.getRowMajorElements().forall(isSafe))`

though, right? (still probably works as written, but does an extra isSafe on the Boolean returned by forall)

**Comment by johnc1231** [comment]
Oh yes duh, thanks. Will fix

**Review by tpoterba — APPROVED** [review]

**Pull request merged**